Add function check_if_any_parent().

jcallaway opened the pull request:
TODO: expand test.

Commit:
Add function check_if_any_parent().

TODO: expand test.

huberthickman commented on check_if_any.R:
```diff
@@ -29,4 +29,18 @@ check_if_any  <- function( participant_codes, safe_query,ncit_code) {
+# Returns true ('YES') if check_if_any would return true for any parent of ncit_code.
+# In other words, starts looking one level up.
+check_if_any_parent <- function(participant_codes, safe_query, ncit_code) {
+  sql <- paste0("select parent from ncit_tc where descendant = $1")
```
I think we need to just look one level up for parents - I think Mike said "look down and one level up".  So the one level down part would be "select parent from ncit_tc_with_path where descendant = $1 and level = 1" -- that gets us just the parents.  (there may be lots of levels above a node that we don't want to roll up to.).

huberthickman commented on check_if_any.R:
```diff
@@ -29,4 +29,18 @@ check_if_any  <- function( participant_codes, safe_query,ncit_code) {
+    }
+  }
+  return('NO')
+}
```
I don't think the semantics here are quite right, and it is a **subtle** thing, and I'll amend my above comment.

Check for descendants OR any direct parent.  
Currently in check_if_any
`sql_tc <-
    paste0(
      "select count(*) as num_descendants from ncit_tc tc where tc.parent = $1 and tc.descendant in ("
      ,
      participant_codes,
      ")"
      )`

the statement in check_if_any_and_parents informally  would be something like
`with 
all_the_matches  as (select count(*) as num_matches from ncit_tc tc where tc.parent = $1 and tc.descendant in ( those participant codes that are splatted in )
      
union	  
select count(*) as num_matches from ncit_tc_with_path tc where tc.descendant = $1 and tc.level =1 and tc.parent in (those participant codes that are splatted in) )
select sum(num_matches) as match_count from all_the_matches 


if match_count > 0 then return YES else return NO`

huberthickman replied:
![image](https://github.com/CBIIT/sec_poc/assets/2655068/f49465f1-fc46-4869-8792-a9a706e57ab0)

huberthickman replied:
This looks right to me now.

Commit:
Feedback from code review.

Only look at direct parents, not their descendants that don't include the code in question.
